rpc/interface/src/rpc_handler.rs

```diff
@@ -0,0 +1,57 @@
+//! RPC handler trait.
+
+//---------------------------------------------------------------------------------------------------- Use
+use std::{future::Future, task::Poll};
+
+use axum::{http::StatusCode, response::IntoResponse};
+use futures::{channel::oneshot::channel, FutureExt};
+use tower::Service;
+
+use cuprate_helper::asynch::InfallibleOneshotReceiver;
+use cuprate_json_rpc::Id;
+use cuprate_rpc_types::json::JsonRpcRequest;
+
+use crate::{rpc_error::RpcError, rpc_request::RpcRequest, rpc_response::RpcResponse};
+
+//---------------------------------------------------------------------------------------------------- RpcHandler
+/// An RPC handler.
+///
+/// This trait represents a type that can turn [`RpcRequest`]s into [`RpcResponse`]s.
+///
+/// Implementors of this trait must be [`tower::Service`]s that use:
+/// - [`RpcRequest`] as the generic `Request` type
+/// - [`RpcResponse`] as the associated `Response` type
+/// - [`RpcError`] as the associated `Error` type
+/// - `InfallibleOneshotReceiver<Result<RpcResponse, RpcError>>` as the associated `Future` type
```

Comment:
I don't think we should put this bound `InfallibleOneshotReceiver` on the implementer as it's quite restrictive and pretty much prevents the use of middleware.